packages/master-detail-layout/src/vaadin-master-detail-layout.js

```diff
@@ -27,26 +27,140 @@ class MasterDetailLayout extends ElementMixin(ThemableMixin(PolylitMixin(LitElem
   static get styles() {
     return css`
       :host {
-        display: block;
+        display: flex;
+        box-sizing: border-box;
+        height: 100%;
       }
 
       :host([hidden]) {
         display: none !important;
       }
+
+      :host(:not([has-detail])) [part='detail'] {
+        display: none;
+      }
+
+      /* No fixed size */
+      :host(:not([has-master-size])) [part='master'] {
+        flex-grow: 1;
+        flex-basis: var(--_master-min-size, 50%);
```

Comment:
This feels weird. If I just add master and detail content they share space 50/50, makes sense. 

Then I set a min size on one of them and the space distribution changes. All I wanted to do is define a min-size at which the layout should switch to overlay mode, not change the distribution ratio.

Reply:
Thanks, I somehow misinterpreted the original idea. The flex-basis should be actually set when both `masterSize` and `detailSize` are set according to Rolf:

> as for the case when both sides have size set, I still think it would be more useful to treat that as
> - both sides flex
> - min-size is determined by the size set on each

It makes sense to move `detailMinSize` and `masterMinSize` to the PR adding overlay mode. I'll update the PR.

Reply:
Yes, at least at this point, I don't see any need for using the min size properties in CSS. We'll only need them in the resize observer for switching modes.